src/Microsoft.DotNet.Interactive/InstallPackagesMessage.cs

```diff
@@ -0,0 +1,105 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.DotNet.Interactive.Formatting;
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Encodings.Web;
+using Microsoft.AspNetCore.Html;
+using static Microsoft.DotNet.Interactive.Formatting.PocketViewTags;
+
+namespace Microsoft.DotNet.Interactive
+{
+    // Data bag containing the data to be displayed when reporting nuget resolve progress 
+    public class InstallPackagesMessage
+    {
+        static InstallPackagesMessage ()
+        {
+            Formatter.Register<InstallPackagesMessage>(m => m.FormatAsPlainText(), PlainTextFormatter.MimeType);
+            Formatter.Register<InstallPackagesMessage>(m => m.FormatAsHtml(), HtmlFormatter.MimeType);
+            Formatter.SetPreferredMimeTypeFor(typeof(InstallPackagesMessage), PlainTextFormatter.MimeType);
+        }
+
+        public IEnumerable<string> RestoreSources;
+        public IEnumerable<string> InstalledPackages;
+        public IEnumerable<string> InstallingPackages;
```

Comment:
Any reason for these to be `IEnumerable<string>` rather than `IReadOnlyList<string>`?

Reply:
The only thing you can do with them is iterate them.  It's pretty much the smallest possible contract.

Reply:
If the underlying implementation is always expected to be non-dynamic / non-lazy, then we can avoid allocations (e.g. `Count` rather than `Count()`) using `IReadOnlyList<string>`. It's just a habit.

Reply:
IReadOnlyList<string> is one of the ugliest type names in existance.